src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -15,6 +16,9 @@ namespace System.Formats.Tar
     // Writes header attributes of a tar archive entry.
     internal sealed partial class TarHeader
     {
+        private const long Octal12ByteFieldMaxValue = (1L << (3 * 11)) - 1; // Max value of 11 octal digits.
+        private const int Octal8ByteFieldMaxValue = (1 << (3 * 7)) - 1;     // Max value of 7 octal digits.
```

Comment:
Maybe we could use these limits in the PosixTarEntry and TarEntry setter exceptions for ModificationTime, DeviceMajor and DeviceMinor, when the tar entry format is not GNU. Thoughts?

The higher value maybe does not need to throw for PAX anymore, as you are storing the value in the extended attributes when it does not fit in the standard field.

Reply:
> The higher value maybe does not need to throw for PAX anymore, as you are storing the value in the extended attributes when it does not fit in the standard field.

Yes, we can also allow the larger range for the PAX format. I'll include this.